runtime/connectors/blob/csvreader.go

```diff
@@ -26,38 +33,44 @@ func downloadCSV(ctx context.Context, bucket *blob.Bucket, obj *blob.ListObject,
 	return err
 }
 
-func csvRows(reader *ObjectReader, option *extractOption) ([]byte, error) {
-	switch option.strategy {
+func csvRows(reader *ObjectReader, option *csvExtractOption) ([]byte, error) {
+	switch option.extractOption.strategy {
 	case runtimev1.Source_ExtractPolicy_STRATEGY_HEAD:
-		return csvRowsHead(reader, option)
+		return csvRowsHead(reader, option.extractOption)
 	case runtimev1.Source_ExtractPolicy_STRATEGY_TAIL:
 		return csvRowsTail(reader, option)
 	default:
-		panic(fmt.Sprintf("unsupported strategy %s", option.strategy))
+		panic(fmt.Sprintf("unsupported strategy %s", option.extractOption.strategy))
 	}
 }
 
-func csvRowsTail(reader *ObjectReader, option *extractOption) ([]byte, error) {
-	header, err := getHeader(reader)
-	if err != nil {
-		return nil, err
+func csvRowsTail(reader *ObjectReader, option *csvExtractOption) ([]byte, error) {
+	header := make([]byte, 0)
+	bytesToRead := option.extractOption.limitInBytes
+	if option.hasHeader {
+		// csv has header, need to read header first
+		headerRow, err := getHeader(reader)
+		if err != nil {
+			return nil, err
+		}
+		headerRow = append(headerRow, _newLineSeparator...)
+		header = headerRow
+		bytesToRead = option.extractOption.limitInBytes - uint64(len(header))
```

Comment:
Remove line 49 and instead pull this line out after the if check – if there's no header, it will just subtract by 0